src/generic/Matrix.jl

```diff
@@ -142,10 +146,35 @@ Base.getindex(V::MatSpaceVecView, i::Int) = V.entries[i]
 
 Base.setindex!(V::MatSpaceVecView{T}, z::T, i::Int) where {T} = (V.entries[i] = z)
 
-Base.setindex!(V::MatSpaceVecView, z::RingElement, i::Int) = setindex!(V.entries, V.base_ring(z), i)
+Base.setindex!(V::MatSpaceVecView, z::NCRingElement, i::Int) = setindex!(V.entries, V.base_ring(z), i)
 
 Base.size(V::MatSpaceVecView) = (length(V.entries), )
 
+Base.length(V::MatSpacePointView) = 1
+
+Base.getindex(V::MatSpacePointView) = V.entries[]
+
+function Base.getindex(V::MatSpacePointView, i::Int)
+  i == 1 || error(BoundsError(V, i))
```

Comment:
```suggestion
  i == 1 || throw(BoundsError(V, i))
```
otherwise you create an `Error` whose message is a printed `BoundsError`. (same another two times below)

But I think I would prefer to leave out this check altogether as the underlying `SubArray` will perform the check again anyway

Reply:
These methods aren't needed at all, as some julia base magic delegates them automatically. Some nice explanation can be found in `?IndexStyle`.